ports/nrf/mpconfigport.h

```diff
@@ -263,10 +267,16 @@ extern const struct _mp_obj_module_t ble_module;
     { MP_ROM_QSTR(MP_QSTR_os), MP_ROM_PTR(&mp_module_uos) }, \
     { MP_ROM_QSTR(MP_QSTR_time), MP_ROM_PTR(&mp_module_utime) }, \
 
+#if MICROPY_VFS
+#define BUILTIN_OPEN { MP_ROM_QSTR(MP_QSTR_open), MP_ROM_PTR(&mp_builtin_open_obj) },
+#else
```

Comment:
I took a look at the minimal port. I guess this is one of the methods that are expected to be found on all platforms. Should we consider to keep the open() compiled in, and add a MICROPY_VFS define in builtin_open.c? The builtin_open.c is not in the Makefile source file list at the moment, but would serve this purpose of selecting between FAT or dummy implementation.

Reply:
Any idea why `open` would exist on platforms without file support? Just for consistency? And what about `os.listdir()`?
Sure, we can keep it in using a `MICROPY_VFS` guard. Note that I would like to add microbit filesystem without VFS support soon (like on the microbit itself), so `open()` may actually be useful on all platforms.

Reply:
Fixed.
I've also removed `builtin_open.c`, as it didn't seem to be referenced anywhere.